### DISCUSSION TITLE
Can CSS be extracted into a file that's statically linked instead of injected?

### DISCUSSION THREAD

**Discussion by Alkalurops** [flareact/flareact#112] [Q&A]
Google is unhappy with the style/content changes when the style is injected after web pack was loaded and started to run…

**Comment by traffisco** [comment]
> Google is unhappy with the style/content changes when the style is injected after web pack was loaded and started to run…

What do you mean unhappy?

**Reply by Alkalurops** [reply]
Google Lighthouse says there are resources blocking the rendering. The reason is, that the CSS gets injected into the head from the web pack, after javascript has been loaded parsed and run…

**Reply by traffisco** [reply]
Hmm, I see them bitchin about that too but the performance penalty is not significant.
For me its a minor difference in load time that not really worth fixing. Very interested to follow this one. Please update if you have new info about this lighthouse situation

**Reply by Alkalurops** [reply]
This unforuntatelly also means: The site is entirely unstyled if JavaScript is off.

**Reply by traffisco** [reply]
Where do you see "CSS gets injected into the head from the web pack, after javascript has been loaded parsed and run"?
In my app, I see that the CSS is already in the head before anything else loaded and Lighthouse shows correct styling. 
Like I said, I do see an issue in Lighthouse but its because I use styled components in part of my app.

**Comment by jplhomer** [comment]
Hi! Flareact supports traditional CSS style sheets embedded in the document. It also automatically compiled styles, minifies them, and includes links to them in the server response. 

Check out flareact.com to see this in action. If you are using something like styled-jsx or another CSS-in-JS framework without properly including the sheets in a custom `_document` page, it's possible you will see styles added with JS in production as you described. 

Check out the docs for more https://flareact.com/docs/built-in-css-support

**Reply by Alkalurops** [reply]
I have an `_app.tsx` that does this:

```typescript
import "./default.scss"
```

and I wanted to write that it will not add the link, but I double checked and it actually does. 

So I'm a bit confused why google reported this. It is cleary not the case anymore.

Thanks for your help & many thanks for this great framework!

**Reply by Alkalurops** [reply]
Ist it possible to include the compiled css directly in the html without injection?
